include/rcutils/strerror.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCUTILS__STRERROR_H_
+#define RCUTILS__STRERROR_H_
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcutils/visibility_control.h"
+
+/// Retrieve the string corresponding to the last system error.
+/* This function retrieves the value of errno, and calls the system-specific
+ * equivalent of `strerror` on it, storing the output in the provided buffer.
+ * If the error message is longer than the buffer, it will be truncated.
+ * The memory for the c-string buffer that is passed in must be managed by the
+ * caller.
+ *
+ * This function is thread-safe.
```

Comment:
Why is this function thread-safe where the underlying platform specific implementation doesn't guarantee to prevent data races?

Reply:
As far as I know, it does guarantee that.  We are using `strerror_r` on Linux/macOS, which is thread-safe.  On Windows, we are using `strerror_s`, which I believe is an implementation of the C11 version of the same name: https://wiki.sei.cmu.edu/confluence/display/c/CON33-C.+Avoid+race+conditions+when+using+library+functions .  This is also guaranteed to be thread-safe.